src/routes/clusters.py

```diff
@@ -84,3 +85,23 @@ async def create_cluster(
     )
 
     return result
+
+
+@router.get(
+    "",
+    response_model=SparkClusterStatus,
+    summary="Get cluster status",
+    description="Retrieves the status of the Spark cluster for the authenticated user.",
+)
+async def get_cluster_status(
+    user: kb_auth.KBaseUser = Depends(auth),
+) -> SparkClusterStatus:
+    """Get the status of the Spark cluster belonging to the authenticated user."""
+
+    manager = KubeSparkManager(
+        username=str(user.user),
+    )
+
+    status = manager.get_cluster_status()
+
+    return status
```

Comment:
Should this return a non 2xx if there's an error?

Reply:
I actually want this get method to always return so that users can always check their clustering status. If we raise here, we will probably need to add another exception in the exception_handler. I feel like just put the error into the response is better and easier.

Reply:
I don't mean raise an exception, I just mean return a 4XX / 5XX with the current body

Reply:
hmmm That would seem a bit off to me. All 4xx/5xx will return `ErrorResponse`. Feels a little strange to me if we include `SparkClusterStatus` in the `ErrorResponse`. 
```
class ErrorResponse(BaseModel):
    """Standard error response model."""

    error: Annotated[int | None, Field(description="Error code")] = None
    error_type: Annotated[str | None, Field(description="Error type")] = None
    message: Annotated[str | None, Field(description="Error message")] = None

```

Reply:
You'd have to override it with a JsonResponse or something, but yeah that's a good point. You don't want to have different error structures. In that case, I'd just make it extremely clear in the endpoint docs that a 200 does not necessarily mean the cluster is in a usable state and they *must* check the error field for the master and workers.

Actually, maybe an `ok` boolean or `error` boolean in the top level doc might be a good idea

Reply:
👍

Reply:
I'd call out that field in the endpoint docs as well to be sure users are aware that they should check it and not assume things are ok because they got a 200, but I leave that up to you

Reply:
👍